src/apps/core/app.js

```diff
@@ -89,13 +90,42 @@ define(function(require) {
 			self.displayFavicon();
 			self.loadSVG();
 
+			if (crossSiteMessaging !== undefined) {
+				self.bindCrossSiteMessagingHandler(crossSiteMessaging);
+			}
+
 			container.append(mainTemplate);
 
 			monster.waterfall([
 				_.bind(self.loadIncludes, self)
 			], _.bind(self.loadAuth, self));
 		},
 
+		bindCrossSiteMessagingHandler: function(crossSiteMessaging) {
+			const { origin, topics } = crossSiteMessaging;
+
+			var handleCrossSiteMessages = function handleCrossSiteMessages(event) {
+				var activeApp = monster.apps.getActiveApp(),
+					triggerApp = event.data.split('.')[0];
```

Comment:
Check if `event.data` is a string, before the split, to avoid errors caused by messages with a different payload. This can be checked by using `lodash#isString`.